app/src/main/java/fr/free/nrw/commons/location/LocationServiceManager.java

```diff
@@ -212,6 +213,14 @@ public void onProviderEnabled(String provider) {
     @Override
     public void onProviderDisabled(String provider) {
         Timber.d("Provider %s disabled", provider);
```

Comment:
Hi @Glitch101, instead of making a new instance of NearbyParentFragment, as the NearbyParentFragmentPresenter is already implementing LocationUpdateListener, how about adding a new method in LocationUpdateListener, enableLocationOffDialog(boolean shouldEnable) and as the presenter will be implementing it, it can be routed from there to the fragment

Reply:
![image](https://user-images.githubusercontent.com/3358282/75967431-b6661b00-5ec3-11ea-923c-a487d792b0ba.png)
[The boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)

Reply:
> ![image](https://user-images.githubusercontent.com/3358282/75967431-b6661b00-5ec3-11ea-923c-a487d792b0ba.png)
> [The boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)

Thanks, @macgills for introducing me to **The boolean trap**. I will take care of it in this PR and in the future.